api/actions/serializers.py

```diff
@@ -306,17 +300,16 @@ def create(self, validated_data):
             raise PermissionDenied('You do not have permission to perform this trigger at this time')
         except ValueError:
             raise PermissionDenied('You do not have permission to perform this trigger at this time')
+        except MachineError:
+            raise PermissionDenied('You do not have permission to perform this trigger at this time')
 
-        target.reload()
-        after_sanction_state = RegistrationModerationStates.from_db_name(target.moderation_state)
-        determined_trigger = RegistrationModerationTriggers.from_transition(
-            prior_sanction_state, after_sanction_state,
-        )
+        target.refresh_from_db()
+        determined_trigger = RegistrationModerationTriggers.from_db_name(target.actions.last().trigger)
         request_trigger = RegistrationModerationTriggers.from_db_name(trigger)
```

Comment:
nit: why `from_db_name` here and in L307? I'd either do this before L280 so you can compare against the enums instead of the `.db_name`s in L281 and 283 or else just compare the strings in L310